parboiled-core/src/main/scala/org/parboiled2/Parser.scala

```diff
@@ -94,16 +94,19 @@ abstract class Parser(initialValueStackSize: Int = 16, maxValueStackSize: Int =
   private var _cursor: Int = _
 
   // the value stack instance we operate on
-  private var _valueStack: ValueStack = _
+  private var _valueStack: ValueStack = new ValueStack(initialValueStackSize, maxValueStackSize)
 
   // the current ErrorAnalysisPhase or null (in the initial run)
   private var phase: ErrorAnalysisPhase = _
 
+  private var _maxLength = -1
+
   def copyStateFrom(other: Parser, offset: Int): Unit = {
     _cursorChar = other._cursorChar
     _cursor = other._cursor - offset
     _valueStack = other._valueStack
     phase = other.phase
+    _maxLength = other._maxLength
```

Comment:
This seems smelly to me.
If the `other` parser has the same `input` then this assignment shouldn't be necessary.
And if it has another input then maybe `copyStateFrom` shouldn't be allowed at all, no?
Maybe, rather than copying over the `maxLength` value, we should `require(input eq other.input)` or sth like that?
Maybe I am misunderstanding the utility of this method though...

Reply:
> And if it has another input then maybe copyStateFrom shouldn't be allowed at all, no?

So judging from the context at https://github.com/apache/incubator-pekko-http/commit/d3b1322d4526171370935ec7b4648bc45fabc672#diff-efb20db992bf7767a1160a413f6a7c51a719c786d8b6593bf6b446ea1a6e9fdbR160 this appears. to be the case. It is also noted that in the original commit that "This is not quite done API-wise.".

If this is the case then it seems like it makes sense to properly think about this?

Reply:
Ok, looking a bit more into this method it looks like it's mainly used for running sub-parsers, which are running on a `__SubParserInput`, which has length `input.length - offset`.
So it seems to me that the correct line here would be `_maxLength = other._maxLength - offset`.

Reply:
I will double check this doesn't case any regressions in pekko http and if it doesn't then I will commit it

Reply:
Okay so at least from pekko-http side this didn't create any regression, change committed

Reply:
Yes, that's my assumption as well.
Still not a great solution here to blindly assign the cached length to some value without seeing or verifying the actual `input`...